packages/SwingSet/src/controller/startXSnap.js

```diff
@@ -62,32 +88,47 @@ export function makeStartXSnap(options) {
     netstringMaxChunkSize: NETSTRING_MAX_CHUNK_SIZE,
   };
 
+  /** @param {StartXSnapLoadDetails | undefined} loadDetails */
+  function getSnapshotLoadOptions(loadDetails) {
+    if (!loadDetails) return undefined;
+
+    if (loadDetails.source === 'snapStore' && snapStore) {
+      const { vatID } = loadDetails;
+      const { hash: snapshotID, snapPos } = snapStore.getSnapshotInfo(vatID);
+      // console.log('startXSnap from', { snapshotID });
+      const snapshotStream = snapStore.loadSnapshot(vatID);
+      const snapshotDescription = `${vatID}-${snapPos}-${snapshotID}`;
+      return { snapshotStream, snapshotDescription };
+    }
+
+    if (loadDetails.source === 'snapshot') {
+      const { snapshotStream, snapshotDescription } = loadDetails;
+      return { snapshotStream, snapshotDescription };
+    }
+
+    return undefined;
+  }
+
   /**
-   * @param {string} vatID
    * @param {string} name
    * @param {object} details
    * @param {import('../types-external.js').BundleID[]} details.bundleIDs
    * @param {(request: Uint8Array) => Promise<Uint8Array>} details.handleCommand
    * @param {boolean} [details.metered]
-   * @param {boolean} [details.reload]
+   * @param {StartXSnapLoadDetails} [details.load]
```

Comment:
I think I'd like this option to include "snapshot" in the name, maybe `defailts.snapshotInfo` or `snapshotLoadInfo`

Reply:
No. This is meant to inform the mode in which the worker is created: from scratch and load supervisor and lockdown bundles, or from snapshot. Ultimately it is meant to be mandatory, but the bundleIDs plumbing was a bit too much to change in this PR.